arangod/Aql/FixedVarExpressionContext.cpp

```diff
@@ -31,22 +31,28 @@
 using namespace arangodb;
 using namespace arangodb::aql;
 
-bool FixedVarExpressionContext::isDataFromCollection(
-    Variable const* variable) const {
-  return false;
-}
-
 AqlValue FixedVarExpressionContext::getVariableValue(Variable const* variable,
                                                      bool doCopy,
                                                      bool& mustDestroy) const {
-  mustDestroy = false;
+  // first check temporary variables in QueryExpressionContext.
+  if (!_variables.empty()) {
+    auto it = _variables.find(variable);
+
+    if (it != _variables.end()) {
+      // copy the slice we found
+      mustDestroy = true;
+      return AqlValue((*it).second);
+    }
+  }
```

Comment:
that's 4th occurrence of this code :-) better to move it to a separate protected member of QueryExpressionContext. That will allow us to make `_variables` member private.

Reply:
Done